kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-common/src/main/java/org/kie/server/remote/rest/common/resource/KieServerRestImpl.java

```diff
@@ -225,5 +231,44 @@ public Response updateReleaseId( @Context HttpHeaders headers,
     public Response getServerState(@Context HttpHeaders headers) {
         return createCorrectVariant(server.getServerState(), headers);
     }
+    
+    
+    
+    @ApiOperation(value="Readiness check for KIE Server that indicates that server is fully booted and ready to accept requests",
+            response=Void.class, code=200)
+    @ApiResponses(value = { @ApiResponse(code = 503, message = "Service not yet available") })
+    @GET
+    @Path("readiness")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response readiness(@Context HttpHeaders headers) { 
+        if (server.isKieServerReady()) {
+            return Response.status(Response.Status.OK).build();
+        }
+        return serviceUnavailable();
```

Comment:
I think all "check/monitoring" methods should have always OK response status and show issues as messages to user in case of error. Might be better if users build monitoring apps against these checks

Reply:
these endpoints are meant for machine reading rather than human - so the most important is to communicate clearly over status codes with minimal payload that's why by default they always return just status and no payload. When user wants to see what is not healthy then there is ?report=true query param that will return payload with messages regardless of the status code (both 200 and 503)